From d7c1fe920a79e812768c957de089c5f71e341ce0 Mon Sep 17 00:00:00 2001
From: David Pacheco <dap@joyent.com>
Date: Fri, 10 Feb 2017 14:57:20 -0800
Subject: [PATCH] MANTA-3160 electric-moray not using bootstrap resolvers
 Reviewed by: Robert Mustacchi <rm@joyent.com> Approved by: Jordan Hendricks
 <jordan.hendricks@joyent.com>

---
 lib/server.js                          | 22 +++++++++++++---------
 sapi_manifests/electric-moray/template |  5 +++++
 2 files changed, 18 insertions(+), 9 deletions(-)

diff --git a/lib/server.js b/lib/server.js
index 1ecf56b..51f7ce0 100644
--- a/lib/server.js
+++ b/lib/server.js
@@ -9,6 +9,7 @@
  */
 
 var assert = require('assert-plus');
+var clone = require('clone');
 var fast = require('fast');
 var fs = require('fs');
 var moray_client = require('moray'); // client
@@ -67,6 +68,7 @@ function createServer(options) {
         log.info('creating moray clients');
         createClient({
             ring: opts.ring,
+            morayOptions: options.morayOptions,
             log: options.log
         }, function (_err, clients) {
             if (_err) {
@@ -724,6 +726,7 @@ function createClient(options, callback) {
     assert.object(options, 'options');
     assert.object(options.log, 'options.log');
     assert.object(options.ring, 'options.ring');
+    assert.object(options.morayOptions, 'options.morayOptions');
     assert.func(callback, 'options.callback');
 
     var log = options.log;
@@ -744,17 +747,18 @@ function createClient(options, callback) {
                 url: pnodeUrl
             }, 'creating moray client');
 
-            var client = moray_client.createClient({
-                unwrapErrors: true,
-                log: options.log.child({
-                    component: 'moray-client-' + pnodeUrl.hostname
-                }),
-                srvDomain: pnodeUrl.hostname,
-                cueballOptions: {
-                    defaultPort: parseInt(pnodeUrl.port, 10)
-                }
+            var morayargs = clone(options.morayOptions);
+            if (!morayargs.cueballOptions) {
+                morayargs.cueballOptions = {};
+            }
+            morayargs.unwrapErrors = true;
+            morayargs.srvDomain = pnodeUrl.hostname;
+            morayargs.cueballOptions.defaultPort = parseInt(pnodeUrl.port, 10);
+            morayargs.log = options.log.child({
+                component: 'moray-client-' + pnodeUrl.hostname
             });
 
+            var client = moray_client.createClient(morayargs);
             clientMap[pnode] = client;
             clientArray.push(client);
 
diff --git a/sapi_manifests/electric-moray/template b/sapi_manifests/electric-moray/template
index 11d5294..813eb65 100644
--- a/sapi_manifests/electric-moray/template
+++ b/sapi_manifests/electric-moray/template
@@ -12,6 +12,11 @@
         "size": 5000,
         "expiry": 60
     },
+    "morayOptions": {
+        "cueballOptions": {
+            "resolvers": [ "nameservice.{{DOMAIN_NAME}}" ]
+        }
+    },
     "ringCfg": {
         "leveldbCfg": {
             "errorIfExists": false,
-- 
2.21.0

